taxis/views.py

```diff
@@ -1,3 +1,80 @@
-from django.shortcuts import render
+import json
+import datetime
 
-# Create your views here.
+from django.http.response   import JsonResponse
+from django.views           import View
+from django.db.models       import Q
+
+from taxis.models           import *
+class TaxiListView(View):
+    def get(self, request):
+        departure_location_name  = request.GET.get('departure_location_name', None)
+        arrival_location_name    = request.GET.get('arrival_location_name', None)
+        seat_type                = request.GET.get('seat_type')
+        seat_remain              = request.GET.get('seat_remain')
+        departure_date_string    = request.GET.get('departure_date')
+        
+        departure_time           = request.GET.get('departure_time', '1900-01-01 23:59')
+        price                    = request.GET.get('price', 60000)
+        taxi_company             = request.GET.getlist('taxi_company', ['Dasul Taxi', 'Taxi Choi-gging', 'Art Transportation', 'Lama 운수', 'DaMo taxi', 'Muy bien Trans'])
+        sort                     = request.GET.get('sort', None)
+
+        sort_list = {
+            'dep_time': '-course__departure_time',
+            'price'   : '-price',
+            None      : '-price'
+        }
+        sort           = None if sort not in sort_list else sort
+        sort_string    = sort_list[sort]
+
+        departure_date = datetime.datetime.strptime(departure_date_string, "%Y-%m-%d")
+
+        courses = Course.objects.filter(
+            Q(departure_location__name=departure_location_name) 
+            & Q(arrival_location__name=arrival_location_name))
+
+
+        print(departure_date, seat_type, seat_remain, price, taxi_company, departure_time, sort, departure_location_name, arrival_location_name)
+
+        schedules = Schedule.objects.filter(
+              Q(date=departure_date) 
+            & Q(seat_type__name=seat_type) 
+            & Q(course_id__in=[course.id for course in courses]) 
+            & Q(seat_remain__gte = seat_remain)
+            & Q(price__lte=price)
+            & Q(course__taxi_company__name__in = taxi_company)
+            & Q(course__departure_time__lte = departure_time)).order_by(sort_string)[0:300]
```

Comment:
```suggestion
            & Q(course__departure_time__lte = departure_time)).order_by(sort_string)
```
[0:300]은 저희가 test을 위해 데이터 수를 줄여 놓은 것 입니다. 그래서 제거해 주시는 것이 맞을 것 같아요

Reply:
네 알겠습니다!